web/src/components/studies/StudyDetails.tsx

```diff
@@ -60,17 +63,61 @@ export default function StudyDetails({ study }: StudyDetailsProps) {
   };
 
   useEffect(() => {
-    const calculateRiskScore = () => {
-      // todo: how to determine if they have data?
+    const fetchAssets = async () => {
+      const studyId = study.id;
+      const assetResponse = await getStudiesByStudyIdAssets({ path: { studyId } });
+      if (assetResponse.response.ok && assetResponse.data) {
+        if (assetResponse.data.length > 0) {
+          return assetResponse.data;
+        } else {
+          return [];
+        }
+      }
+    };
+    const calculateRiskScore = async () => {
       let score = 0;
+      setRiskScoreLoading(true);
```

Comment:
This useEffect feels a bit too long. Could you separate the logic into abstracted functions to make it more readable?

Reply:
https://github.com/ucl-arc-tre/portal/pull/344/commits/204fb61ab96c63a3b48cc38bf2fedbe774e657d9 ?